pages/accountLists/[accountListId]/reports/goalCalculator/[[...contactId]].page.tsx

```diff
@@ -65,13 +62,13 @@ const GoalCalculatorContent: React.FC<GoalCalculatorContentProps> = ({
         <RightPanelTitle variant="h6">{t('Details')}</RightPanelTitle>
         <IconButton
           size="small"
-          onClick={() => toggleRightPanel()}
+          onClick={closeRightPanel}
           aria-label={t('Close Panel')}
         >
           <CloseIcon fontSize="small" />
         </IconButton>
       </RightPanelHeader>
-      <RightPanelContent>{rightPanelStepComponent}</RightPanelContent>
+      <RightPanelContent>{rightPanelContent}</RightPanelContent>
```

Comment:
Lets put this in a useMemo or separate component so it doesn't re-create on each render

Reply:
Are you asking to memoize the `rightPanel` JSX content? We don't need to memoize JSX because React recreates components' JSX on every render. If we put it in a separate component, it will still recreate the JSX on every render. That's just how the React rendering process works. But please let me know if I'm missing something or misunderstanding you.